guide-how-to-write-apis.md

```diff
@@ -0,0 +1,339 @@
+| `GET`    | Used to retrieve a single item or a collection of items. |
+| `POST`   | Used when creating new items e.g. a new user, post, comment etc. |
+| `PATCH`  | Used to update one or more fields on an item e.g. update e-mail of user. |
+| `PUT`    | Used to replace a whole item (all fields) with new data. |
+| `DELETE` | Used to delete an item.                  |
+
+
+### Examples
+
+Now that we’ve learned about the anatomy of our endpoints and the different request methods that we should use, it’s time for some examples:
+
+| Method   | URL                                      | Description                              |
+| -------- | ---------------------------------------- | ---------------------------------------- |
+| `GET`    | `/api/posts`                             | Retrieve all posts.                      |
+| `POST`   | `/api/posts`                             | Create a new post.                       |
+| `GET`    | `/api/posts/28`                          | Retrieve post #28.                       |
+| `PATCH`  | `/api/posts/28`                          | Update data in post #28.                 |
+| `POST`   | `/api/posts/comments`                    | Add comment to post #28.                 |
+| `GET`    | `/api/posts/comments/?status=approved&limit=10&page=4` | Retrieve comments for post #28 which are approved. But only 10 per page - starting from page 4. |
+| `DELETE` | `/api/comment/1987`                      | Delete comment #1987.                    |
+| `GET`    | `/api/users/?active=true&sort=username&direction=asc&search=nodes` | Search for “nodes” in active users, sorted  by username ascendingly. |
+
+
+#### Tips on what NOT to do
+
+Hopefully you’re already aware of this. But not at any given time should your endpoint end with `.json`, `.xml` or `.something`. If in doubt, reach out to any of the backend developers at Nodes.
+
+
+## Headers
+
+We love headers! And we have a few that we use pretty much in every one of our endpoints. It’s also a very nice way to send information along with your request without “parameter bomb” your endpoint, e.g. language or 3rd party tokens.
+
+Below is a list of our most used headers. You might end up working on a project where you have to integrate with this weird 3rd party API, which requires you to make a custom header that is not listed below. Go for it, Batman. No requirements here.
+
+| Header key        | Description                              |
+| ----------------- | ---------------------------------------- |
+| `Accept`          | This header is **required by all endpoints**. It’s used to identify the request as our own and to versioning our endpoints. **Default value**: `application/vnd.nodes.v1+json`. |
+| `Authorization`   | Should contain the authorized user’s token. This is used to gain access to protected endpoint. |
+| `Language`        | Should contain ISO 639 code of language translations should be returned in. |
```

Comment:
Browsers already set an `Accept-Language` header:
`Accept-Language:en-US,en;q=0.8,da;q=0.6`

This is also the header which Translate currently uses (or the `X-Accept-Language` header)